modyn/supervisor/internal/triggers/datadrifttrigger.py

```diff
@@ -200,16 +238,32 @@ def _run_detection(self) -> tuple[bool, dict[str, MetricResult]]:
 
         # Compute embeddings
         assert self.embedding_encoder is not None
+
+        # tbd. reuse the embeddings as long as the reference window is not updated
         reference_embeddings = get_embeddings(self.embedding_encoder, reference_dataloader)
         current_embeddings = get_embeddings(self.embedding_encoder, current_dataloader)
         reference_embeddings_df = convert_tensor_to_df(reference_embeddings, "col_")
         current_embeddings_df = convert_tensor_to_df(current_embeddings, "col_")
 
         drift_results = {
-            **self.evidently_detector.detect_drift(reference_embeddings_df, current_embeddings_df),
-            **self.alibi_detector.detect_drift(reference_embeddings, current_embeddings),
+            **self.evidently_detector.detect_drift(reference_embeddings_df, current_embeddings_df, is_warmup),
+            **self.alibi_detector.detect_drift(reference_embeddings, current_embeddings, is_warmup),
         }
+
+        # make the final decisions with the decision engines
+        for metric_name, metric_result in drift_results.items():
+            distance = (
+                metric_result.distance
+                if isinstance(metric_result.distance, float)
+                else float(mean(metric_result.distance))
+            )
+            drift_results[metric_name].is_drift = self.decision_engines[metric_name].evaluate_decision(
```

Comment:
I think one part that I find confusing in naming is that we have drift _detectors_. Detectors detect drift and return a binary is_drift variable. However now we don't necessarily  use the `is_drift` result. I think the separation of concerns is not clear yet. For example:

https://github.com/eth-easl/modyn/blob/0d4217642fd04607a1c02d7cdd5ab35b689980fc/modyn/supervisor/internal/triggers/drift/detector/alibi.py#L70C17-L70C25

here we implement the threshold check. 

we reimplement the check then in the decision engine. 

I think we want to refactor the interactions between detector, aggregation logic, and the decision engine.

Reply:
I find it very unlikely that we will ever use the drift decisions from the detectors, as hypothesis testing is a dead end. The threshold comparison in `alibi.py` is a simple attempt to get the same interface as we get from evidently. With all my offline exploration in mind, I'd propose to remove all the decision support from the detectors (so they will only serve the distance computation --> rename to e.g. `DistanceDevice`?). Then, we could strip all the hypothesis configuration support. Wdyt?